src/main/java/me/jellysquid/mods/sodium/client/render/chunk/ChunkRenderManager.java

```diff
@@ -183,7 +190,7 @@ private void addChunkToRenderLists(ChunkRenderContainer<T> render) {
             T state = states[i];
```

Comment:
We should hoist `this.renderContext` into a local variable to help the JVM optimize here (as the field can be null). For example:

```java
RenderContext<?> ctx = this.renderContext;
// ...

if (state != null) {
  ChunkRenderList<T> list = ctx.chunkRenderLists[i];
// ...
```